module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp

```diff
@@ -111,8 +113,10 @@ namespace localisation {
                     arma::vec3 rTFf   = {s.position[0], s.position[1], 0};
                     Hft.translation() = rTFf;
                     Hft.rotateZ(s.heading);
-                    Hfw = Hft * Htw;
-                    states.push_back(transform3DToFieldState(Hfw));
+                    Hfw      = Hft * Htw;
+                    auto tmp = transform3DToFieldState(Eigen::Affine3d(convert(Hfw)));
+                    states.push_back(Transform2D(convert(Eigen::Vector2d(tmp.translation())),
+                                                 Eigen::Rotation2Dd(tmp.linear()).angle()));
```

Comment:
What?

Reply:
I'll add arma back to `transform.hpp` to fix this